### DISCUSSION TITLE
What is the default server streaming timeout (xhr.timeout) and how to modify it?

### DISCUSSION THREAD

**Discussion by kanekv** [grpc/grpc-web#1375] [Q&A - General]

**Comment by sampajano** [comment]
@kanekv Hi :)

I think you might be able to specify the timeout in Envoy config: [example](https://github.com/grpc/grpc-web/blob/master/net/grpc/gateway/examples/echo/envoy.yaml), if that's the proxy you're using.

**Comment by kanekv** [comment]
@sampajano Client is terminating the stream, there is no timeout on the server, if I connect from another client stream can last forever. From the browser it is terminated within 5-8 minutes. I wonder where this timeout is set.

**Reply by sampajano** [reply]
Aha ok thanks for clarifying!

I'm guess you're not setting any deadline here?
https://github.com/grpc/grpc-web#setting-deadline

As far as i understand from reading our code, the client should respect the above-mentioned deadline, but otherwise doesn't try to specify a XHR timeout:

https://github.com/grpc/grpc-web/blob/49d3b7086895de22b44ec7be29b4c259b553bff8/javascript/net/grpc/web/grpcwebclientbase.js#L326-L345

Maybe you could try specifying the [deadline](https://github.com/grpc/grpc-web#setting-deadline) and see if it's honored? it might help us narrow down the search.

**Reply by kanekv** [reply]
@sampajano that seems to be for unary calls. How do we do that for server streaming?

**Reply by sampajano** [reply]
> @sampajano that seems to be for unary calls. How do we do that for server streaming?

Ah right. Although this is the only occurrence in our codebase that calls `xhr.setTimeoutInterval` so i don't see how we're doing it for server streaming also.

> Client is terminating the stream, there is no timeout on the server, if I connect from another client stream can last forever.

I wonder how did you determine that client is terminating the stream?

If you connect from another grpc client, you'd likely not be using grpc-web, and also not Envoy. And that's why i suspected that Envoy was terminating the request originally.
